CMakeLists.txt

```diff
@@ -18,7 +18,7 @@ project(
 
 # Set the default build type for single-config generators.
 # The build type variable is still overridable from outside.
-set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "Default build type for single-config generators")
+set(CMAKE_BUILD_TYPE_INIT "Debug" CACHE STRING "Default build type for single-config generators")
```

Comment:
[Docs](https://cmake.org/cmake/help/latest/manual/cmake-variables.7.html):

> Note CMake reserves identifiers that:
>
> * begin with CMAKE&#x5f; (upper-, lower-, or mixed-case), or
> * begin with &#x5f;CMAKE&#x5f; (upper-, lower-, or mixed-case), or
> * begin with &#x5f; followed by the name of any [CMake Command](https://cmake.org/cmake/help/latest/manual/cmake-commands.7.html#manual:cmake-commands(7)).

This variable is unknown to CMake and it uses a reserved identifier.

Reply:
It is the variable that determines the initial value of `CMAKE_BUILD_TYPE` when not specified, so CMake knows it. After a quick research, I've discovered that is an `internal` variable and not intended for the end-user. I'll revert this change on my next commit.